Source/RxSwift/once.swift

```diff
@@ -25,12 +25,15 @@ extension Observable {
 	*/
 
 	public static func once(_ element: Element) -> Observable<Element> {
-		var delivered: UInt32 = 0
+        let lock = NSRecursiveLock()
+		var isDelivered = false
 		return create { observer in
-			let wasDelivered = OSAtomicOr32OrigBarrier(1, &delivered)
-			if wasDelivered == 0 {
+            lock.lock()
+			if !isDelivered {
```

Comment:
I don't think you need this `if` statement anymore. Am I wrong about this?

Reply:
Maybe I missed the point, but who could I guarantee that event was only dispatched once without `if` statement?

Reply:
Hey @freak4pc could u please clarify? :)

Reply:
Hey, sorry. This looks OK, can you just fix up the formatting? locking seems to be non-indented, etc

Reply:
Sorry, uses spaces instead of tabs...
Fixed